clusterloader2/GETTING_STARTED.md

```diff
@@ -0,0 +1,227 @@
+- Create single node cluster using [Kind]
+- Implement a simple Clusterloader2 test and run it
+- Run load test on 100 nodes cluster
+
+## Clone perf-test repository
+
+Start with cloning perf-test repository:
+```bash
+git clone git@github.com:kubernetes/perf-tests.git
+cd perf-tests
+```
+
+## Install GVM
+Follow instructions on [GVM install].
+Install golang with specific version (1.15.12 was tested in this tutorial):
+```bash
+gvm install go1.15.12
+gvm use go1.15.12
+```
+Next, add perf-test repository to GOPATH:
+
+```bash
+gvm linkthis k8s.io/perf-tests
+```
+
+## Create cluster using kind
+Follow kind installation guide [Kind install]
+
+Create cluster v1.21.1 with one master and one node:
+```bash
+kind create cluster --image=kindest/node:v1.21.1 --name=test-cluster --wait=5m
+```
+
+Check that you can connect to cluster:
+```bash
+kubectl get nodes
+```
+
+This command generates cluster access credentials, which are stored in `${HOME}/.kube/config` with context name test-cluster.
+
+## Prepare simple test to run
+Let's prepare our first test config (config.yaml).
+This test will:
+- Create one namespace
+- Create single deployment with 1 pod
+- Measure startup latency of this pod
+```yaml
+# Test name
+name: test
+# Number of namespaces that will be created
+namespace:
+  number: 1
+# Tuning set with qps 1 will execute 1 action every second.
+# In out case, creating deployment will be 1 action to execute.
+tuningSets:
+- name: Uniform1qps
+  qpsLoad:
+    qps: 1
+# Test consists of list of steps. A step can be either collection of Phases or Measurements.
+# A Phase defines a state the cluster should reach.
+# A Measurement allows to measure something or wait for something.
+# You can find list of available measurements here: https://github.com/kubernetes/perf-tests/tree/master/clusterloader2#measurement
+steps:
+# We want to start two measurement.
+# One will measure time it takes to start a pod, second will make sure we wait for pod to be in running state.
+# Setting the action field to "start" begins execution of measurement.
+- name: Start measurements
+  measurements:
+  - Identifier: PodStartupLatency
+    Method: PodStartupLatency
+    Params:
+      action: start
+# This label specifies for which pods we want to measure pod startup latency.
+      labelSelector: group = test-pod
+# If pod startup latency goes over threshold, test will fail.
+      threshold: 5s
+  - Identifier: WaitForControlledPodsRunning
+    Method: WaitForControlledPodsRunning
+    Params:
+      action: start
+      apiVersion: apps/v1
+# Type of object we are waiting for
+      kind: Deployment
+      labelSelector: group = test-pod
+# How long we want to wait till object is in running state.
+      operationTimeout: 120s
+# Once we created these two measurements, we can create deployment in next step.
+- name: Create deployment
+  phases:
+# namespaceRange specifies in which namespaces we want to create this deployment.
+  - namespaceRange:
```

Comment:
I've always wanted to default to "all" if you don't specify it...

Reply:
Documentation says that it's considered cluster scope if you don't specify it.